### PR TITLE
feat(node): add TS plugins support for build executor

### LINKED ISSUE #2147: support for nestjs compiler plugins (e.g. for swagger)
I would like to use a nestjs compiler plugin (e.g. described in https://trilon.io/blog/nestjs-swagger-4-whats-new for generation of swagger definition)

How can I do this with nx?

### PR DESCRIPTION
**Pull request by nartc**
## Current Behavior

TS Plugins aren't supported out of the box. Workarounds have been utilizing custom webpack configuration.

## Expected Behavior

TS Plugins are supported out of the box.

A plugin is specified as a valid TS Plugin is an object that has either one of the following: `before`, `after`, or `afterDeclarations` with the following shape:

> `options` are the plugin custom configuration

```ts
export interface CompilerPlugin {
  before?: (
    options?: Record<string, unknown>,
    program?: ts.Program
  ) => Transformer;
  after?: (
    options?: Record<string, unknown>,
    program?: ts.Program
  ) => Transformer;
  afterDeclarations?: (
    options?: Record<string, unknown>,
    program?: ts.Program
  ) => Transformer;
}
```

## Related Issue(s)

Fixes #2147

### CODE REVIEW

File: packages/node/src/utils/types.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import ts = require('typescript');
```
Comment (FrozenPandaz) [inline packages/node/src/utils/types.ts:1]: We should be able to use `import type { } from 'typescript';` so this file doesn't bring in `typescript` (which is heavy).

**Review by FrozenPandaz — APPROVED** [review]

File: packages/node/src/utils/types.ts (hunk 1 above)
Comment (nartc) [reply packages/node/src/utils/types.ts]: Changes made. Thanks Jason

**Review by AgentEnder — APPROVED** [review]

File: packages/node/src/utils/node.config.spec.ts [hunk 2]
```diff
@@ -1,15 +1,15 @@
 import { join } from 'path';
+import { getNodeWebpackConfig } from './node.config';
+import TsConfigPathsPlugin from 'tsconfig-paths-webpack-plugin';
+import { BuildNodeBuilderOptions } from './types';
+
```
Comment (gioragutt) [inline packages/node/src/utils/node.config.spec.ts:5]: Doesn't this change nullify the mocks done below?
AFAIK jest still cannot mock with `import`s, so you have to import/require after the mocks.

Any reason this is not true here?

**Review by gioragutt — COMMENTED** [review]
Great PR, can't wait for it to go in 🚀

File: packages/node/src/utils/node.config.spec.ts (hunk 2 above)
Comment (AgentEnder) [reply packages/node/src/utils/node.config.spec.ts:5]: Jest should hoist the mocks iirc (see https://jestjs.io/docs/manual-mocks#using-with-es-module-imports)

Comment (nartc) [reply packages/node/src/utils/node.config.spec.ts:5]: Like @AgentEnder mentioned, Jest hoists the mocks for you.

**Review by gioragutt — APPROVED** [review]

**Review by gioragutt — APPROVED** [review]

**Review by gioragutt — APPROVED** [review]

**Review by gioragutt — APPROVED** [review]

**Pull request merged**